### PR TITLE
Various warning fixes

### PR DESCRIPTION
**Pull request by lrusak**
We seem to be gaining some new warnings so this PR intends to fix a bunch of them and some old warnings.

7e23ef80fe2a3b4774d00ea30adeecac742a777a: fixes a warning introduced in 4dc197a915aaf9fc98e8db3186f80b52012dd728
c9ff5c73448379a3261ee2ee097ff370e229ebe6: fixes a warning introduced in 6326e7bbb674bb98823948d55e782055d69e1b75
cf3ec214284bb7fa7bc6a7da556ac85216fded3e: fixes warnings introduced in 6326e7bbb674bb98823948d55e782055d69e1b75
8779b95ab5cd596f38b4b0be2d054e893c5c9670: fixes warnings introduced in 67f0f91ad7b9ea8e5bd1342536eba65f1f6ed638

I've also run clang-format on xbmc-xrandr.c completely because it's a mess and would mess up my commits.
I fixed all the `double-promotion` and `sign-compare` warnings in xbmc-xrandr.c

### CODE REVIEW

File: xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp [hunk 1]
```diff
@@ -193,11 +230,11 @@ void CAEBitstreamPacker::PackTrueHD(CAEStreamInfo &info, uint8_t* data, int size
     }
   }
 
-  for (nextCodeIdx = 0; nextCodeIdx < ARRAY_SIZE(MatCodes); nextCodeIdx++)
+  for (nextCodeIdx = 0; nextCodeIdx < static_cast<int>(MatCodes.size()); nextCodeIdx++)
```
Comment (thexai) [inline xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp:233]: Header `Util.h` can be removed now as only was used for `ARRAY_SIZE` macro

File: xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp [hunk 2]
```diff
@@ -22,6 +22,43 @@ extern "C"
...
+{
+  int pos;
+  const uint8_t* code;
+  unsigned int len;
+};
+
+std::array<MatCode, 3> MatCodes = {{
+    {0, mat_start_code.data(), mat_start_code.size()},
+    {30708, mat_middle_code.data(), mat_middle_code.size()},
+    {MAT_FRAME_SIZE - sizeof(mat_end_code), mat_end_code.data(), mat_end_code.size()},
```
Comment (thexai) [inline xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp:57]: Maybe replace `sizeof(mat_end_code)` by `mat_end_code.size()`

File: xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp:232]: Thanks! Updated.

File: xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp (hunk 2 above)
Comment (lrusak) [reply xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp]: Thanks! Updated.

**Review by thexai — APPROVED** [review]
Runtime tested Windows x64 (tested TrueHD passthrough)

**Comment by lrusak** [comment]
> Runtime tested Windows x64 (tested TrueHD passthrough)

Thanks for testing!

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
anyone else want to have a look?

**Comment by lrusak** [comment]
last chance to review 🙃

**Review by garbear — APPROVED** [review]

**Pull request merged**